src/PublicKeyRegistry.sol

```diff
@@ -17,6 +17,8 @@ contract PublicKeyRegistry {
     /// @notice Raised when the public key of the given user has already been set
     error AlreadySet();
 
+
+    /// Q(danielw): it seems if an incorrect pubkey is set, then there is no way to correct it.
```

Comment:
Correct. The rationale behind this being:
- The derived wallet generation is an automated process, with no human intervention
- One wallet can only generate one derived wallet, given the same signing payload
- If someone registers an incorrect public key, that would be very, very strange. This would mean that proposals could be read (potentially) by someone else, other than the signer.

If such an edge case arises, my recommendation would be to make it obvious to the rest of Security Council members, when the bad actor sees that he/she needs to start using another wallet and asks the multisig to replace the address.

The UI only encrypts proposal's data for public keys registered by multisig signers, but without adding ZK proofs, there's no way to verify that the public key of each signer was generated using the private key of their wallet. 

We can bypass this check and allow anyone to change their public key, but updating keys might become a non-feature that opens the door to members silently sharing data with someone else, without needing to expose their personal wallet.

Reply:
Works for me.